src/python/test/test_extract_expression.py

```diff
@@ -1111,11 +1129,91 @@ def run_dx_extract_assay_expression_cmd(self, dataset_or_cohort, filters_json, a
         if additional_fields:
             command.extend(["--additional-fields", additional_fields])
 
-        process = subprocess.check_output(
-            command, shell=True, universal_newlines=True,
-        )
+        if extra_args:
+            command.append(extra_args)
+
+        if subprocess_run:
+            process = subprocess.run(
+                command, capture_output=True, text=True, check=False
+            )
+
+        else:
+            process = subprocess.check_output(
+                command,
+                universal_newlines=True,
+            )
+
         return process
 
+    def test_dx_extract_cmd_location_expression_sample_sql(self):
+        expected_sql_query = "SELECT `expression_1`.`feature_id` AS `feature_id`, `expression_1`.`sample_id` AS `sample_id`, `expression_1`.`value` AS `expression`, `expr_annotation_1`.`gene_name` AS `feature_name`, `expr_annotation_1`.`chr` AS `chrom`, `expr_annotation_1`.`start` AS `start` FROM `database_gzky7400vgpyzy621q43gkkf__molecular_expression1_db`.`expression` AS `expression_1` LEFT OUTER JOIN `database_gzky7400vgpyzy621q43gkkf__molecular_expression1_db`.`expr_annotation` AS `expr_annotation_1` ON `expression_1`.`feature_id` = `expr_annotation_1`.`feature_id` WHERE (`expr_annotation_1`.`chr` = '11' AND (`expr_annotation_1`.`start` BETWEEN 8693350 AND 67440200 OR `expr_annotation_1`.`end` BETWEEN 8693350 AND 67440200 OR `expr_annotation_1`.`start` <= 8693350 AND `expr_annotation_1`.`end` >= 67440200) OR `expr_annotation_1`.`chr` = 'X' AND (`expr_annotation_1`.`start` BETWEEN 148500700 AND 148994424 OR `expr_annotation_1`.`end` BETWEEN 148500700 AND 148994424 OR `expr_annotation_1`.`start` <= 148500700 AND `expr_annotation_1`.`end` >= 148994424) OR `expr_annotation_1`.`chr` = '17' AND (`expr_annotation_1`.`start` BETWEEN 75228160 AND 75235759 OR `expr_annotation_1`.`end` BETWEEN 75228160 AND 75235759 OR `expr_annotation_1`.`start` <= 75228160 AND `expr_annotation_1`.`end` >= 75235759)) AND `expression_1`.`value` >= 25.63 AND `expression_1`.`sample_id` IN ('sample_1', 'sample_2')"
```

Comment:
maybe you could add that to src/python/test/expression_test_assets/expression_test_expected_output_dict.py

Reply:
Thanks! Yep, I'll do this after the spec update.